src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/rc_command_request.cc

```diff
@@ -0,0 +1,251 @@
+                 "Remote control is not supported by system");
+    return;
+  }
+  LOG4CXX_TRACE(logger_, "RC interface is available!");
+  if (CheckDriverConsent()) {
+    if (AcquireResources()) {
+      Execute();  // run child's logic
+    }
+    // If resource is not aqcuired, AcquireResources method will either
+    // send response to mobile or
+    // send additional request to HMI to ask driver consent
+  }
+}
+
+bool RCCommandRequest::AcquireResources() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  const std::string module_type =
+      (*message_)[strings::msg_params][message_params::kModuleType].asString();
+
+  if (!IsResourceFree(module_type)) {
+    LOG4CXX_WARN(logger_, "Resource is busy.");
+    SendResponse(false, mobile_apis::Result::IN_USE, "");
+    return false;
+  }
+
+  AcquireResult::eType acquire_result = AcquireResource(message_);
+  switch (acquire_result) {
+    case AcquireResult::ALLOWED: {
+      SetResourceState(module_type, ResourceState::BUSY);
+      return true;
+    }
+    case AcquireResult::IN_USE: {
+      SendResponse(false, mobile_apis::Result::IN_USE, "");
+      return false;
+    }
+    case AcquireResult::ASK_DRIVER: {
+      SetResourceState(module_type, ResourceState::BUSY);
+      SendGetUserConsent(module_type);
+      return false;
+    }
+    case AcquireResult::REJECTED: {
+      SendResponse(false, mobile_apis::Result::REJECTED, "");
+      return false;
+    }
+  }
+  return false;
+}
+
+void RCCommandRequest::on_event(const app_mngr::event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  const std::string module_type =
+      (*message_)[strings::msg_params][message_params::kModuleType].asString();
+
+  SetResourceState(module_type, ResourceState::FREE);
+
+  if (event.id() == hmi_apis::FunctionID::RC_GetInteriorVehicleDataConsent) {
+    ProcessAccessResponse(event);
+  }
+}
+
+void RCCommandRequest::ProcessAccessResponse(
+    const app_mngr::event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  app_mngr::ApplicationSharedPtr app =
+      application_manager_.application(CommandRequestImpl::connection_key());
+  const std::string module_type =
+      (*message_)[strings::msg_params][message_params::kModuleType].asString();
+  if (!app) {
+    LOG4CXX_ERROR(logger_, "NULL pointer.");
+    SendResponse(false, mobile_apis::Result::APPLICATION_NOT_REGISTERED, "");
+    return;
+  }
+
+  const smart_objects::SmartObject& message = event.smart_object();
+
+  mobile_apis::Result::eType result_code =
+      GetMobileResultCode(static_cast<hmi_apis::Common_Result::eType>(
+          message[app_mngr::strings::params][app_mngr::hmi_response::code]
+              .asUInt()));
+
+  const bool result =
+      helpers::Compare<mobile_apis::Result::eType, helpers::EQ, helpers::ONE>(
+          result_code,
+          mobile_apis::Result::SUCCESS,
+          mobile_apis::Result::WARNINGS);
+
+  bool is_allowed = false;
+  if (result) {
+    if (true ==
```

Comment:
@LitvinenkoIra avoid true/false comaparations in ifs:
`if(x)` is better then `if(x==true)`